src/System.Private.CoreLib/src/System/Threading/WaitHandle.Windows.cs

```diff
@@ -12,29 +12,29 @@ namespace System.Threading
 {
     public abstract partial class WaitHandle
     {
-        internal static unsafe int WaitForSingleObject(IntPtr handle, int millisecondsTimeout)
+        internal static unsafe int WaitForSingleObject(IntPtr handle, int millisecondsTimeout, bool interruptible)
         {
             SynchronizationContext context = RuntimeThread.CurrentThread.SynchronizationContext;
-            bool useSyncContextWait = (context != null) && context.IsWaitNotificationRequired();
+            bool useSyncContextWait = interruptible && (context != null) && context.IsWaitNotificationRequired();
```

Comment:
It would be slightly better to put an if(!interruptible) around the whole sync context stuff, it's unnecessary to get the current sync context for uninterruptible waits